github/copilot_test.go

```diff
@@ -0,0 +1,649 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+// Test invalid JSON responses, vlaid responses are covered in the other tests
+func TestCopilotSeatDetails_UnmarshalJSON(t *testing.T) {
+	tests := map[string]struct {
+		data    string
+		want    *CopilotSeatDetails
+		wantErr bool
+	}{
+		"Invalid JSON": {
+			data: `{`,
+			want: &CopilotSeatDetails{
+				Assignee: nil,
+			},
+			wantErr: true,
+		},
+		"Invalid Assignee Field Type": {
+			data: `{
+					"assignee": "test"
+				}`,
+			want:    &CopilotSeatDetails{},
+			wantErr: true,
+		},
+		"Invalid Assignee Type": {
+			data: `{
+					"assignee": {
+						"name": "octokittens",
+						"id": 1,
+						"type": "None"
+					}
+				}`,
+			want:    &CopilotSeatDetails{},
+			wantErr: true,
+		},
+		"Invalid User": {
+			data: `{
+					"assignee": {
+						"type": "User",
+						"id": "bad",
+					}
+				}`,
+			want:    &CopilotSeatDetails{},
+			wantErr: true,
+		},
+		"Invalid Team": {
+			data: `{
+					"assignee": {
+						"type": "Team",
+						"id": "bad",
+					}
+				}`,
+			want:    &CopilotSeatDetails{},
+			wantErr: true,
+		},
+		"Invalid Organization": {
+			data: `{
+					"assignee": {
+						"type": "Organization",
+						"id": "bad",
+					}
+				}`,
+			want:    &CopilotSeatDetails{},
+			wantErr: true,
+		},
+	}
+
+	for name, tc := range tests {
+		seatDetails := &CopilotSeatDetails{}
+
+		t.Run(name, func(t *testing.T) {
+			err := seatDetails.UnmarshalJSON([]byte(tc.data))
```

Comment:
It's better to test with `json.Unmarshal` instead of directly calling `UnmarshalJSON` because that is how users will call it.